src/Microsoft.AspNetCore.TestHost/TestServer.cs

```diff
@@ -27,6 +27,14 @@ public TestServer(IWebHostBuilder builder)
             _hostInstance = host;
         }
 
+        public TestServer(IWebHostBuilder builder, Action<TestServer> configureServer)
+        {
+            configureServer(this);
```

Comment:
Hmm, I see. And I can't move `configureServer` down below initialization of host because by then `Build` has already been called.  I guess the configureServer solution won't work then.  I wanted to make it a function because I wanted to keep it flexible to accommodate more needs than just my own issue.

I'm imagining `TestSeverOptions` as a bunch of flags to which map to one off cases and I don't think that is scalable.

Was imagining something like:
```
public class TestServerOptions {
  public bool AddServerAddressesFeature { get; set; }
}
...
public TestServer(IWebHostBuilder builder, TestServerOptions testServerOptions) {
  if (testServerOptions.AddServerAddressesFeatures) {
    var server = this as IServer;
    server.Features.Set<IServerAddressesFeature>(new ServerAddressesFeature);
  }
}
```
Maybe you were imaging something different, do you have example?  If it is something like the above, this would likely lead TestServerOptions having many flags added to it for unrelated purposes.

I pushed new change which takes in `IFeatureCollection` and copies them into the internal collection.  I think it's clear about use case and doesn't expose `this` like old iteration.


```
public class TestServerOptions {
  public bool AddServerAddresses { get; set; }
}
...
public